status/raw/status-app-protocols.md

```diff
@@ -0,0 +1,372 @@
+- `content_topic` MUST be set to the app-level content topic
+- `timestamp` MUST be set to the current Unix epoch timestamp (in nanosecond precision)
+
+### Pubsub topics and sharding
+
+All Waku messages are published to pubsub topics as defined in [23/WAKU2-TOPICS](../../waku/informational/23/topics.md).
+Since pubsub topics define a routing layer for messages,
+they can be used to shard traffic.
+The pubsub topic used for publishing a message depends on the app-level [functional scope](#functional-scope).
+
+**Self-addressed messages**
+
+The application MUST define at least one distinct pubsub topic for self-addressed messages.
+The application MAY defined a set of more than one pubsub topic for self-addressed messages to allow traffic sharding for scalability.
+
+**Global messages**
+
+The application MUST define at least one distinct pubsub topic for global control messages and global content messages.
+The application MAY defined a set of more than one pubsub topic for global messages to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Community messages**
+
+The application SHOULD define at least one separate pubsub topic for each separate community's community control and community content messages.
+The application MAY define a set of more than one pubsub topic per community to allow traffic sharding for scalability.
+It is RECOMMENDED that separate pubsub topics be used for global control messages and global content messages.
+
+**Large messages**
+
+The application MAY define separate pubsub topics for large messages.
+These pubsub topics for large messages MAY be distinct for each functional scope.
+
+### Resource usage
+
+The application can use a range of Waku protocols to interact with the Waku transport layer.
+The specific set of Waku protocols used depend on desired functionality and resource usage profile for the specific client.
+Resources can be restricted in terms of bandwidth and computing resources.
+
+Waku protocols that are more appropriate for resource-restricted environments are often termed "light protocols".
+Waku protocols that consume more resources, but simultaneously contribute more to Waku infrastructure, are often termed "full protocols".
+The terms "full" and "light" is just a useful abstraction than a strict binary, though,
+and Status clients can operate along a continuum of resource usage profiles,
+each using the combination of "full" and "light" protocols most appropriate to match its environment and motivations.
+
+To simplify interaction with the selection of "full" and "light" protocols,
+Status clients MUST define a "full mode" and "light mode"
+to allow users to select whether their client would prefer "full protocols" or "light protocols" by default.
+Status Desktop clients are assumed to have more resources available and SHOULD use full mode by default.
```

Comment:
> I think for user facing devices/apps it should be light by default (desktop could have a switch to enable full mode). In most users' experience, the same app in the different devices should mostly behave the same, there may be more features in a desktop environment, but by default we should make it consistent between devices.

Ultimately the question is for Status product team. But considering the desired properties, it is fair to assume that a desktop app may consume more resources. The same way it's fair to assume that using tor browser is slower, or using BitTorrent over FTP consumes more upload bandwidth.

If the Status product team decides that Desktop and Mobile must behave the same, and both must use light mode by default. Then the question is "where is the infrastructure"?

This can be compensated by either:
1.  encouraging users to run companion "nodes": a second software on your laptop, rasberry pi, cloud or enable relay mode
3.  Or having incentivization in place, where users or community owners pay for the resources of the network, with a large enough number of nodes to support this relay network

In this context, it also mean that by "default" privacy and censorship resistance are less than what they could be.

(1) is unlikely to be enough to allow the application to be reliable, as regularly stated.

(2) we are not ready yet, work is in progress and I expect to agressivly ramp up this stream in 2025 H1.

Once (2) is ready, then Status product will have a choice between:
A. Free product to use, but users share resources by default
B. Paying product for users, but they dont share resources by default.

(A) was clearly the selected direction so far by Status.